### PR TITLE
add nix icon

### PR DESCRIPTION
**Pull request by o-az**
#### Description

Adds Nix icon

- Closes # 
- What does this PR change? Give us a brief description.
- Did you change something visual? A before/after screenshot can be helpful.

### CODE REVIEW

**Comment by at-the-vr** [comment]
This [Discussion thread](https://github.com/withastro/starlight/discussions/1599) is worth mentioning. Other than that I believe this will require a changeset 💡

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -120,6 +120,8 @@ export const BuiltInIcons = {
...
+	nix:
+		'<path d="m7.352 1.592l-1.364.002L5.32 2.75l1.557 2.713l-3.137-.008l-1.32 2.34h11.69l-1.353-2.332l-3.192-.006l-2.214-3.865zm6.175 0l-2.687.025l5.846 10.127l1.341-2.34l-1.59-2.765l2.24-3.85l-.683-1.182h-1.336l-1.57 2.705l-1.56-2.72zm6.887 4.195l-5.846 10.125l2.696-.008l1.601-2.76l4.453.016l.682-1.183l-.666-1.157l-3.13-.008L21.778 8.1l-1.365-2.313zM9.432 8.086l-2.696.008l-1.601 2.76l-4.453-.016L0 12.02l.666 1.157l3.13.008l-1.575 2.71l1.365 2.315zM7.33 12.25l-.006.01l-.002-.004l-1.342 2.34l1.59 2.765l-2.24 3.85l.684 1.182H7.35l.004-.006h.001l1.567-2.698l1.558 2.72l2.688-.026l-.004-.006h.01zm2.55 3.93l1.354 2.332l3.192.006l2.215 3.865l1.363-.002l.668-1.156l-1.557-2.713l3.137.008l1.32-2.34z"/>',
```
Comment (delucis) [inline packages/starlight/components/Icons.ts:124]: I think we can squish this down a little more without visual impact thanks to [SVGOMG](https://jakearchibald.github.io/svgomg/):

```suggestion
		'<path d="M7.4 1.6H6l-.7 1.1L7 5.5H3.7L2.4 7.8h11.7l-1.3-2.3H9.6l-2.2-4zm6.1 0h-2.7l5.9 10.1L18 9.4l-1.6-2.8 2.3-3.8-.7-1.2h-1.3L15 4.3l-1.6-2.7zm7 4.2-6 10.1h2.8l1.6-2.8h4.4L24 12l-.7-1.2h-3.1l1.6-2.7-1.4-2.3zM9.4 8H6.6L5 10.8H.7L0 12l.7 1.2h3.1l-1.6 2.7 1.4 2.3zm-2.2 4.2L6 14.6l1.6 2.8-2.3 3.8.7 1.2h1.3L9 19.7l1.6 2.7h2.7zm2.6 3.9 1.3 2.3h3.2l2.2 3.9H18l.7-1.2-1.6-2.7h3.2l1.3-2.3z"/>',
```

**Review by delucis — COMMENTED** [review]
Thank you for the PR @o-az! As @at-the-vr noted, we have had requests for this already and I think Nix makes sense to include.

Left one suggestion to optimise the path a little more for smaller file size. And I can take care of adding a “changeset” (which is how we organise our change logs and releases).

**Comment by o-az** [comment]
Thank you @delucis for taking the time to review. I accepted your suggestion.

**Comment by Eveeifyeve** [comment]
Mentioned in discussion.

**Pull request merged**